lib/internal/repl.js

```diff
@@ -92,7 +92,15 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
 
   function oninit(err, hnd) {
     if (err) {
-      return ready(err);
+      // Cannot open history file.
+      // Don't crash, just don't persist history.
+      debug('oninit: %s', err.message);
```

Comment:
If this is reached we already know what is happening, I suggest moving this after the error message and doing `debug(err.stack);` like above.